src/sqlfluff/utils/reflow/reindent.py

```diff
@@ -27,3 +36,792 @@ def deduce_line_indent(raw_segment: RawSegment, root_segment: BaseSegment) -> st
+    # Set up the default anchor
+    anchor = {"before": elements[indent_points[0].idx + 1].segments[0]}
+
+    # Find initial indent, and deduce appropriate string indent.
+    # We also account for potential templated indents.
+    indent_seg = None
+    if indent_points[-1].last_line_break_idx:
+        indent_seg = cast(
+            ReflowPoint, elements[indent_points[-1].last_line_break_idx]
+        )._get_indent_segment()
+    elif isinstance(elements[0], ReflowPoint):
+        # No last_line_break_idx, but this is a point. It's the first line.
+        # Get the last whitespace element.
+        # TODO: We don't currently handle the leading swallowed whitespace case.
+        # That could be added here, but it's an edge case which can be done
+        # at a later date easily. For now it won't get picked up.
+        for indent_seg in elements[0].segments[::-1]:
+            if indent_seg.is_type("whitespace"):
+                break
+        # Handle edge case of no whitespace, but with newline.
+        if not indent_seg.is_type("whitespace"):
+            indent_seg = None
+        else:
+            NotImplementedError(
+                "Cannot find whitespace in leading point. Report this as a bug."
+            )
+
+    if indent_seg:
+        # We have to check pos marker before checking is templated.
+        # Insertions don't have pos_markers - so aren't templated,
+        # but also don't support calling is_templated.
+        if indent_seg.is_type("placeholder"):
+            # It's a consumed indent.
+            current_indent = (
+                cast(TemplateSegment, indent_seg).source_str.split("\n")[-1] or ""
+            )
+        elif not indent_seg.pos_marker or not indent_seg.is_templated:
+            current_indent = indent_seg.raw
+        else:  # pragma: no cover
+            # It's templated. This shouldn't happen. Segments returned by
+            # _get_indent_segment, should be valid indents (i.e. whitespace
+            # or placeholders for consumed whitespace). This is a bug.
+            raise NotImplementedError(
+                "Unexpected templated indent. Report this as a bug on GitHub."
+            )
+    else:
+        current_indent = ""
+
+    # First handle starting indent.
+    assert "\n" not in current_indent, f"Found newline in indent: {indent_seg}"
+    desired_indent_units = indent_line.desired_indent_units(forced_indents)
+    desired_starting_indent = desired_indent_units * single_indent
+    initial_point = cast(ReflowPoint, elements[indent_points[0].idx])
+
+    if current_indent != desired_starting_indent:
+        reflow_logger.debug(
+            "    Correcting indent @ line %s. Existing indent: %r -> %r",
+            elements[indent_points[0].idx + 1].segments[0].pos_marker.working_line_no,
+            current_indent,
+            desired_starting_indent,
+        )
+        # Initial point gets special handling it it has no newlines.
```

Comment:
The word "it" is duplicated. Did you mean this?
```suggestion
        # Initial point gets special handling if it has no newlines.
```